pkg/common/cns-lib/volume/manager.go

```diff
@@ -557,6 +557,11 @@ func (m *defaultManager) DeleteVolume(ctx context.Context, volumeID string, dele
 		}
 		volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 		if volumeOperationRes.Fault != nil {
+			_, isNotFoundFault := volumeOperationRes.Fault.Fault.(*vim25types.NotFound)
```

Comment:
If we receive any other MangedObjectNotFound during delete call, we should not pass the delete volume operation.

What if during delete we receive ManagedObjectNotfound for datastore.

> 2021-03-05T14:51:25.160Z error vsanvcmgmtd[47863] [vSAN@6876 sub=CnsAuthorize opId=38e82535] CNS AuthZ: CNS AuthZ:Failed to retrieve datastore moref for vol 3eb4d733-4880-46a6-b230-12144d17c55f: N3Vim5Fault8NotFound9ExceptionE(Fault cause: vim.fault.NotFound

Reply:
Can we also handle ManagedObjectNotFound for above check at line 533

```
task, err := m.virtualCenter.CnsClient.DeleteVolume(ctx, cnsVolumeIDList, deleteDisk)
		if err != nil {
			if cnsvsphere.IsNotFoundError(err) {
				log.Infof("VolumeID: %q, not found. Returning success for this operation since the volume is not present", volumeID)
				return nil
			}
			log.Errorf("CNS DeleteVolume failed from the  vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
			return err
}
```